miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/NoteRestController.java

```diff
@@ -70,41 +84,208 @@ public void setText(String text) {
-  public void addNote(@PathVariable(name = "entityType") String entityType,
-      @PathVariable(name = "entityId") long entityId,
-      @RequestBody NoteRequest request) throws IOException {
-    addNote(serviceForEntityType(entityType), entityId, request);
+  public void deleteNotes(
+      @PathVariable String entityType,
+      @PathVariable long entityId,
+      @PathVariable String noteIds) throws IOException {
+
+    NoteService<? extends Identifiable> service = serviceForEntityType(entityType);
+    Identifiable entity = service.get(entityId);
+
+    if (entity == null) {
+      throw new RestException(entityType + " not found", Status.NOT_FOUND);
+    }
+
+    List<Long> noteIdList = Arrays.stream(noteIds.split(","))
+        .map(Long::parseLong)
+        .collect(Collectors.toList());
+
+    @SuppressWarnings("unchecked")
+    NoteService<Identifiable> typedService = (NoteService<Identifiable>) service;
+
+    for (Long noteId : noteIdList) {
+      typedService.deleteNote(entity, noteId);
+    }
   }
 
-  private <T extends Identifiable> void deleteNote(NoteService<T> service, long entityId, long noteId)
-      throws IOException {
-    T entity = service.get(entityId);
-    service.deleteNote(entity, noteId);
+  public static class BulkDeleteRequest {
+    private String entityType;
+    private long entityId;
+    private List<Long> ids;
+
+    public String getEntityType() {
+      return entityType;
+    }
+
+    public void setEntityType(String entityType) {
+      this.entityType = entityType;
+    }
+
+    public long getEntityId() {
+      return entityId;
+    }
+
+    public void setEntityId(long entityId) {
+      this.entityId = entityId;
+    }
+
+    public List<Long> getIds() {
+      return ids;
+    }
 
+    public void setIds(List<Long> ids) {
+      this.ids = ids;
+    }
   }
 
-  @DeleteMapping(value = "/{entityType}/{entityId}/{noteId}")
+  @PostMapping("/bulk-delete")
   @ResponseStatus(HttpStatus.NO_CONTENT)
-  public void deleteNote(@PathVariable(name = "entityType") String entityType,
-      @PathVariable(name = "entityId") long entityId,
-      @PathVariable(name = "noteId") long noteId) throws IOException {
-    deleteNote(serviceForEntityType(entityType), entityId, noteId);
+  public void bulkDeleteNotes(@RequestBody BulkDeleteRequest request) throws IOException {
+    NoteService<? extends Identifiable> service = serviceForEntityType(request.getEntityType());
+    Identifiable entity = service.get(request.getEntityId());
+
+    if (entity == null) {
+      throw new RestException(request.getEntityType() + " not found", Status.NOT_FOUND);
+    }
+
+    @SuppressWarnings("unchecked")
+    NoteService<Identifiable> typedService = (NoteService<Identifiable>) service;
+
+    for (Long noteId : request.getIds()) {
+      typedService.deleteNote(entity, noteId);
+    }
+  }
+
+
+  private List<NoteDto> getNotesList(String entityType, long entityId, boolean includeRelated) throws IOException {
```

Comment:
Can make this much simpler and avoid all the bad casting and type checking by checking `entityType` at the start and selecting the appropriate service yourself. If it's a library, use `libraryService.get` directly to get the library, `library.getNotes` to get its notes, `library.getParent` to get its parent sample, etc. Throw an exception if `entityType` isn't sample/library since those are the only types this is expected to handle.